Cant split openapi yaml file into seperate files

### Expected Behavior

OpenApi file can be spitted into separated files, eg:

openapi.yaml
```yaml
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Sample
  description: Multi-file for OpenAPI Specification.
paths:
    /api/customer/{id}/files:
    $ref: "./paths/files.yml"
```

paths/files.yml
```yaml
post:
  tags:
    - Customer
  operationId: uploadFile
  summary: Upload customer files
  description: Save file in customer file repository
  parameters:
    - name: id
      in: path
      required: true
      description: Unique identifier of the Customer
      schema:
        type: string
        format: uuid
  requestBody:
    required: true
    content:
      application/json:
        schema:
          $ref: '../schemas/files.yml#/FileCreateDto'

  responses:
    201:
      description: File uploaded correctly
      content:
        application/json:
          schema:
            $ref: '../schemas/files.yml#/FileId'
```

OpenAPi is correct, compile with cli works fine:

```
swagger-cli bundle openapi.yml --outfile _build/openapi.yaml --type yaml
```

 but on compile I got error



### Actual Behaviour


```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.692 s
[INFO] Finished at: 2024-08-27T08:49:49+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.micronaut.maven:micronaut-maven-plugin:4.6.1:generate-openapi-server (default-generate-openapi-server) on project fk-service: Execution default-generate-openapi-server of goal io.micronaut.maven:micronaut-maven-plugin:4.6.1:generate-openapi-server failed: Range [23, 17) out of bounds for length 17 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.micronaut.maven:micronaut-maven-plugin:4.6.1:generate-openapi-server (default-generate-openapi-server) on project fk-service: Execution default-generate-openapi-server of goal io.micronaut.maven:micronaut-maven-plugin:4.6.1:generate-openapi-server failed: Range [23, 17) out of bounds for length 17
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-generate-openapi-server of goal io.micronaut.maven:micronaut-maven-plugin:4.6.1:generate-openapi-server failed: Range [23, 17) out of bounds for length 17
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:133)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: java.lang.StringIndexOutOfBoundsException: Range [23, 17) out of bounds for length 17
    at jdk.internal.util.Preconditions$1.apply (Preconditions.java:55)
    at jdk.internal.util.Preconditions$1.apply (Preconditions.java:52)
    at jdk.internal.util.Preconditions$4.apply (Preconditions.java:213)
    at jdk.internal.util.Preconditions$4.apply (Preconditions.java:210)
    at jdk.internal.util.Preconditions.outOfBounds (Preconditions.java:98)
    at jdk.internal.util.Preconditions.outOfBoundsCheckFromToIndex (Preconditions.java:112)
    at jdk.internal.util.Preconditions.checkFromToIndex (Preconditions.java:349)
    at java.lang.String.checkBoundsBeginEnd (String.java:4865)
    at java.lang.String.substring (String.java:2834)
    at java.lang.String.substring (String.java:2807)
    at io.micronaut.openapi.generator.MicronautInlineModelResolver.flattenPaths (MicronautInlineModelResolver.java:136)
    at io.micronaut.openapi.generator.MicronautInlineModelResolver.flatten (MicronautInlineModelResolver.java:123)
    at io.micronaut.openapi.generator.AbstractMicronautJavaCodegen.preprocessOpenAPI (AbstractMicronautJavaCodegen.java:743)
    at org.openapitools.codegen.DefaultGenerator.configureGeneratorProperties (DefaultGenerator.java:295)
    at org.openapitools.codegen.DefaultGenerator.generate (DefaultGenerator.java:1289)
    at io.micronaut.openapi.generator.MicronautCodeGeneratorEntryPoint.generate (MicronautCodeGeneratorEntryPoint.java:118)
    at io.micronaut.maven.openapi.AbstractOpenApiMojo.execute (AbstractOpenApiMojo.java:182)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
```

I tried to set `<micronaut.openapi.additional.files>src/main/resources/openapi</micronaut.openapi.additional.files>` but it not hepled. 

### Steps To Reproduce

split single openapi file and put references into main file

### Environment Information

- Linux Mint
- Java 21
- MN parrent: 4.5.1

```
    <micronaut.openapi.generate.server>true</micronaut.openapi.generate.server>
    <micronaut.openapi.definition>src/main/resources/openapi/openapi.yml</micronaut.openapi.definition>
    <micronaut.openapi.api.package.name>dev.itrust.fk.api</micronaut.openapi.api.package.name>
    <micronaut.openapi.model.package.name>dev.itrust.fk.api.model</micronaut.openapi.model.package.name>
    <micronaut.openapi.invoker.package.name>dev.itrust.fk</micronaut.openapi.invoker.package.name>
    <micronaut.openapi.use.reactive>false</micronaut.openapi.use.reactive>
    <micronaut.openapi.server.use.auth>false</micronaut.openapi.server.use.auth>
    <micronaut.openapi.additional.files>src/main/resources/openapi</micronaut.openapi.additional.files>
```

### Example Application

_No response_

### Version

4.5.1

--- a/openapi-generator/src/main/java/io/micronaut/openapi/generator/MicronautCodeGeneratorEntryPoint.java
+++ b/openapi-generator/src/main/java/io/micronaut/openapi/generator/MicronautCodeGeneratorEntryPoint.java
@@ -85,8 +85,10 @@ public final class MicronautCodeGeneratorEntryPoint {
      * Performs code generation.
      */
     public void generate() {
+        var opts = new ParseOptions();
+        opts.setResolve(true);
         var openAPI = new OpenAPIParser()
-            .readLocation(definitionFile.toString(), null, new ParseOptions()).getOpenAPI();
+            .readLocation(definitionFile.toString(), null, opts).getOpenAPI();
 
         // Configure codegen
         withPath(outputDirectory, codeGenerator::setOutputDir);

--- a/openapi-generator/src/main/java/io/micronaut/openapi/generator/MicronautInlineModelResolver.java
+++ b/openapi-generator/src/main/java/io/micronaut/openapi/generator/MicronautInlineModelResolver.java
@@ -138,7 +138,7 @@ public final class MicronautInlineModelResolver {
 
         for (Map.Entry<String, PathItem> pathsEntry : paths.entrySet()) {
             PathItem path = pathsEntry.getValue();
-            if (path.get$ref() != null) {
+            if (path.get$ref() != null && path.get$ref().startsWith("#/components/pathItems/")) {
                 path = openAPI.getComponents().getPathItems().get(path.get$ref().substring("#/components/pathItems/".length()));
             }
             var operationsMap = new LinkedHashMap<>(path.readOperationsMap());
@@ -183,7 +183,7 @@ public final class MicronautInlineModelResolver {
             String key = responsesEntry.getKey();
             ApiResponse response = responsesEntry.getValue();
 
-            if (response.get$ref() != null) {
+            if (response.get$ref() != null && response.get$ref().startsWith("#/components/responses/")) {
                 response = openAPI.getComponents().getResponses().get(response.get$ref().substring("#/components/responses/".length()));
             }
 
@@ -261,7 +261,7 @@ public final class MicronautInlineModelResolver {
                 continue;
             }
 
-            if (parameter.get$ref() != null) {
+            if (parameter.get$ref() != null && parameter.get$ref().startsWith("#/components/parameters/")) {
                 if (openAPI.getComponents().getParameters() != null) {
                     parameter = openAPI.getComponents().getParameters().get(parameter.get$ref().substring("#/components/parameters/".length()));
                 }
@@ -312,7 +312,7 @@ public final class MicronautInlineModelResolver {
             }
 
             Schema parameterSchema = null;
-            if (parameter.get$ref() != null) {
+            if (parameter.get$ref() != null && parameter.get$ref().startsWith("#/components/parameters/")) {
                 if (openAPI.getComponents().getParameters() != null) {
                     var param = openAPI.getComponents().getParameters().get(parameter.get$ref().substring("#/components/parameters/".length()));
                     parameterSchema = param.getSchema();

--- a/openapi-generator/src/test/java/io/micronaut/openapi/generator/JavaMicronautClientCodegenTest.java
+++ b/openapi-generator/src/test/java/io/micronaut/openapi/generator/JavaMicronautClientCodegenTest.java
@@ -703,4 +703,45 @@ class JavaMicronautClientCodegenTest extends AbstractMicronautCodegenTest {
                 );
             """);
     }
+
+    @Test
+    void testGenerateByMultipleFiles() {
+
+        var codegen = new JavaMicronautClientCodegen();
+        String outputPath = generateFiles(codegen, "src/test/resources/3_0/multiple/swagger.yml", CodegenConstants.APIS, CodegenConstants.MODELS);
+        String path = outputPath + "src/main/java/org/openapitools/";
+
+        assertFileContains(path + "api/CustomerApi.java",
+            """
+                    @Post("/api/customer/{id}/files")
+                    Mono<HttpResponse<@NotNull String>> uploadFile(
+                        @PathVariable("id") @NotNull UUID id,
+                        @Body @NotNull @Valid FileCreateDto fileCreateDto
+                    );
+                """);
+        assertFileContains(path + "model/FileCreateDto.java",
+            """
+                public class FileCreateDto {
+                
+                    public static final String JSON_PROPERTY_TYPE_CODE = "typeCode";
+                    public static final String JSON_PROPERTY_ORG_NAME = "orgName";
+                
+                    /**
+                     * Customer type ORG
+                     */
+                    @NotNull
+                    @Pattern(regexp = "^ORG$")
+                    @JsonProperty(JSON_PROPERTY_TYPE_CODE)
+                    private String typeCode = "ORG";
+                
+                    @NotNull
+                    @JsonProperty(JSON_PROPERTY_ORG_NAME)
+                    private String orgName;
+                
+                    public FileCreateDto(String typeCode, String orgName) {
+                        this.typeCode = typeCode;
+                        this.orgName = orgName;
+                    }
+                """);
+    }
 }

--- a/openapi-generator/src/test/java/io/micronaut/openapi/generator/KotlinMicronautClientCodegenTest.java
+++ b/openapi-generator/src/test/java/io/micronaut/openapi/generator/KotlinMicronautClientCodegenTest.java
@@ -747,4 +747,33 @@ class KotlinMicronautClientCodegenTest extends AbstractMicronautCodegenTest {
                 ): Mono<SuccessResetPassword>
             """);
     }
+
+    @Test
+    void testGenerateByMultipleFiles() {
+
+        var codegen = new KotlinMicronautClientCodegen();
+        String outputPath = generateFiles(codegen, "src/test/resources/3_0/multiple/swagger.yml", CodegenConstants.APIS, CodegenConstants.MODELS);
+        String path = outputPath + "src/main/kotlin/org/openapitools/";
+
+        assertFileContains(path + "api/CustomerApi.kt",
+            """
+                    @Post("/api/customer/{id}/files")
+                    fun uploadFile(
+                        @PathVariable("id") @NotNull id: UUID,
+                        @Body @NotNull @Valid fileCreateDto: FileCreateDto
+                    ): Mono<HttpResponse<String>>
+                """);
+        assertFileContains(path + "model/FileCreateDto.kt",
+            """
+                data class FileCreateDto(
+                    @field:NotNull
+                    @field:Pattern(regexp = "^ORG$")
+                    @field:JsonProperty(JSON_PROPERTY_TYPE_CODE)
+                    var typeCode: String = "ORG",
+                    @field:NotNull
+                    @field:JsonProperty(JSON_PROPERTY_ORG_NAME)
+                    var orgName: String,
+                ) {
+                """);
+    }
 }

--- a/openapi-generator/src/test/resources/3_0/multiple/paths/files.yml
+++ b/openapi-generator/src/test/resources/3_0/multiple/paths/files.yml
@@ -1,0 +1,28 @@
+post:
+  tags:
+    - Customer
+  operationId: uploadFile
+  summary: Upload customer files
+  description: Save file in customer file repository
+  parameters:
+    - name: id
+      in: path
+      required: true
+      description: Unique identifier of the Customer
+      schema:
+        type: string
+        format: uuid
+  requestBody:
+    required: true
+    content:
+      application/json:
+        schema:
+          $ref: '../schemas/files.yml#/FileCreateDto'
+
+  responses:
+    201:
+      description: File uploaded correctly
+      content:
+        application/json:
+          schema:
+            $ref: '../schemas/files.yml#/FileId'

--- a/openapi-generator/src/test/resources/3_0/multiple/schemas/files.yml
+++ b/openapi-generator/src/test/resources/3_0/multiple/schemas/files.yml
@@ -1,0 +1,18 @@
+FileCreateDto:
+  type: object
+  required:
+    - typeCode
+    - orgName
+  description: Organization customer data
+  properties:
+    typeCode:
+      type: string
+      pattern: ^ORG$
+      description: Customer type ORG
+      default: ORG
+      example: ORG
+    orgName:
+      type: string
+
+FileId:
+  type: string

--- a/openapi-generator/src/test/resources/3_0/multiple/swagger.yml
+++ b/openapi-generator/src/test/resources/3_0/multiple/swagger.yml
@@ -1,0 +1,8 @@
+openapi: "3.0.0"
+info:
+  version: 1.0.0
+  title: Swagger Sample
+  description: Multi-file for OpenAPI Specification.
+paths:
+  /api/customer/{id}/files:
+    $ref: "./paths/files.yml"

